framework/db/pgsql/QueryBuilder.php

```diff
@@ -187,17 +187,8 @@ public function resetSequence($tableName, $value = null)
      */
     public function checkIntegrity($check = true, $schema = '', $table = '')
     {
-        $enable = $check ? 'ENABLE' : 'DISABLE';
-        $schema = $schema ? $schema : $this->db->getSchema()->defaultSchema;
-        $tableNames = $table ? [$table] : $this->db->getSchema()->getTableNames($schema);
-        $viewNames = $this->db->getSchema()->getViewNames($schema);
-        $tableNames = array_diff($tableNames, $viewNames);
-        $command = '';
-
-        foreach ($tableNames as $tableName) {
-            $tableName = '"' . $schema . '"."' . $tableName . '"';
-            $command .= "ALTER TABLE $tableName $enable TRIGGER ALL; ";
-        }
+        $enable = $check ? 'IMMEDIATE' : 'DEFERRED';
+        $command = "SET CONSTRAINTS ALL $enable; ";
```

Comment:
This produces unexpcted behavior, since triggers are disabled for all tables, while they should be only for the particular one.
We can not accept such change in the core.